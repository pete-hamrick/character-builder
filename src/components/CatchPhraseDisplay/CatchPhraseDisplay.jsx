import React from 'react'

export default function CatchPhraseDisplay({
  catchphrases,
  hatCount,
  headCount,
  shirtCount,
  bottomCount,
}) {
  return (
    <section>
      <p>
        You have changed the hat {handleCountPhrasing(hatCount)}, the head{' '}
        {handleCountPhrasing(headCount)}, the shirt {handleCountPhrasing(shirtCount)}, and the
        bottom {handleCountPhrasing(bottomCount)}. Of course, no one can forget your characters
        classic catchphrases:
      </p>
      {catchphrases.map((catchphrase) => (
        <p key={catchphrase}>{catchphrase}</p>
      ))}
    </section>
  )
}

const handleCountPhrasing = (currentCount) => {
  if (currentCount === 1) {
    return currentCount + ' time'
  } else {
    return currentCount + ' times'
  }
}
