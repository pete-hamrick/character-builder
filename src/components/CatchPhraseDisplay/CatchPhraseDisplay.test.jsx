import { render, screen } from '@testing-library/react'
import CatchPhraseDisplay from './CatchPhraseDisplay'
import React from 'react'

it('ought to display the CatchPhrase component', () => {
  const { container } = render(
    <CatchPhraseDisplay
      catchphrases={['oh yeah', 'beautiful']}
      hatCount={1}
      headCount={1}
      shirtCount={1}
      bottomCount={1}
    />
  )

  expect(container).toMatchSnapshot()
})
