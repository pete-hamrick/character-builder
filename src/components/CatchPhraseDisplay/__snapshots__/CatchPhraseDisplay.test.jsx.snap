// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ought to display the CatchPhrase component 1`] = `
<div>
  <section>
    <p>
      You have changed the hat 
      1 time
      , the head
       
      1 time
      , the shirt 
      1 time
      , and the bottom 
      1 time
      . Of course, no one can forget your characters classic catchphrases:
    </p>
    <span>
      oh yeah
    </span>
    <br />
    <span>
      beautiful
    </span>
    <br />
  </section>
</div>
`;
